packages/api/src/external/shared/patient-demographics.ts

```diff
@@ -0,0 +1,211 @@
+      state: a.state.trim().toLowerCase() as string,
+      zip: a.zip.trim(),
+      country: a.country?.trim().toLowerCase() ?? undefined,
+    };
+  });
+  const driversLicenses = (patient.data.personalIdentifiers ?? []).flatMap(p => {
+    if (p.type !== "driversLicense") return [];
+    return { value: p.value.trim(), state: p.state.trim().toLowerCase() };
+  });
+  const ssns = (patient.data.personalIdentifiers ?? []).flatMap(p => {
+    if (p.type !== "ssn") return [];
+    return [p.value.trim()];
+  });
+  return {
+    dob,
+    gender,
+    names,
+    telephoneNumbers,
+    emails,
+    addresses,
+    driversLicenses,
+    ssns,
+  };
+}
+
+export function createAugmentedPatient(
+  patient: Patient,
+  linksDempgraphics: LinkDemoData[]
+): Patient {
+  const patientDemographics = patientToLinkedDemoData(patient);
+  const hashedPatientAddresses = patientDemographics.addresses.map(createAddressString);
+  // TODO Is submitting lower case addresses to CQ / CW okay, or do we have to grab the original version?
+  const newAddresses: Address[] = linksDempgraphics.flatMap(ld => {
+    return ld.addresses
+      .filter(a => !hashedPatientAddresses.includes(createAddressString(a)))
+      .map(a => {
+        const lineSplit = (a.line ?? "").split(addressSeparator);
+        return {
+          addressLine1: lineSplit[0] ?? "",
+          addressLine2: lineSplit[1] ? lineSplit.slice(1).join(" ") : undefined,
+          city: a.city ?? "",
+          state: (a.state ?? "") as USState,
+          zip: a.zip ?? "",
+          country: a.country,
+        };
+      });
+  });
+  const newTelephoneNumbers: Contact[] = linksDempgraphics.flatMap(ld => {
+    return ld.telephoneNumbers
+      .filter(tn => !patientDemographics.telephoneNumbers.includes(tn))
+      .map(phone => {
+        return { phone };
+      });
+  });
+  const newEmails: Contact[] = linksDempgraphics.flatMap(ld => {
+    return ld.emails
+      .filter(e => !patientDemographics.emails.includes(e))
+      .map(email => {
+        return { email };
+      });
+  });
+  const aupmentedPatient = {
+    ...patient,
+    data: {
+      ...patient.data,
+      contact: patient.data.contact
+        ? [...patient.data.contact, ...newTelephoneNumbers, ...newEmails]
+        : [...newTelephoneNumbers, ...newEmails],
+      address: [...patient.data.address, ...newAddresses],
+    },
+  };
+  return aupmentedPatient;
+}
+
+function createAddressString(a: LinkDemoDataAddress) {
+  return `${a.line ?? ""} ${a.city ?? ""} ${a.state ?? ""} ${a.zip ?? ""} ${a.country ?? ""}`;
+}
+
+export function linkHasNewDemographicData(
+  patientDemographics: LinkDemoData,
+  linkDemographics: LinkDemoData
+): boolean {
+  // Address
+  // TODO Move hashedPatientAddresses to wrapper function to call once.
+  const hashedPatientAddresses = patientDemographics.addresses.map(createAddressString);
+  const hashedLinkAddresses = linkDemographics.addresses.map(createAddressString);
```

Comment:
i dont see any hashing in either of these. are you sure thats the appropriate name?